onnx/backend/test/cpp/driver/test_driver.cc

```diff
@@ -0,0 +1,232 @@
+#include "test_driver.h"
+#include <cstdio>
+#include <cstdlib>
+#include <fstream>
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#ifdef _WIN32
+#include <io.h>
+#include <windows.h>
+#else
+#include <dirent.h>
+#endif
+
+namespace ONNX_NAMESPACE {
+namespace testing {
+bool FileExists(const std::string& filename) {
+#ifdef _WIN32
+  if (INVALID_FILE_ATTRIBUTES == GetFileAttributes(filename.c_str())) {
+    return false;
+  }
+#else
+  struct stat stats;
+  if (lstat(filename.c_str(), &stats) != 0 || !S_ISREG(stats.st_mode)) {
+    return false;
+  }
+#endif
+  return true;
+}
+
+void TestDriver::SetDefaultDir(const std::string& s) {
+  default_dir_ = s;
+}
+
+/**
+ *	It is possible that case_dir is not a dir.
+ *	But it does not affect the result.
+ */
+void TestDriver::FetchSingleTestCase(const std::string& case_dir) {
+  std::string model_name = case_dir;
+  model_name += "model.onnx";
+  if (FileExists(model_name)) {
+    TestCase test_case;
+    test_case.model_filename_ = model_name;
+    test_case.model_dirname_ = case_dir;
+    for (int case_count = 0;; case_count++) {
+      std::vector<std::string> input_filenames, output_filenames;
+      std::string input_name, output_name;
+      std::string case_dirname = case_dir;
+      case_dirname += "test_data_set_" + ONNX_NAMESPACE::to_string(case_count);
+
+      output_name = case_dirname + "/output_" + "0" + ".pb";
+      if (!FileExists(output_name) && !FileExists(input_name)) {
```

Comment:
input_name is empty?

Reply:
Oh at first I only used output_name to judge, so I didnt add input_name here. I will fix it.